### PR TITLE
Dataloader checking

### PR DESCRIPTION
**Pull request by ChenglongWang**
- Moved datachecking-related functions to `check_data_utils.py`.
- Extract `check_dataloader` function to enable checking dataloader. 
- Minor bugs and typing issues fixed.

Signed-off-by: Chenglong Wang <ryuu.j.ching@gmail.com>

### CODE REVIEW

**Comment by ChenglongWang** [comment]
Potential bugs in check_data_utils.py Line 170 and Line 216 need to be fixed. @ldxFAIRYTAIL 
Line 180 caused crash when drawing contour.

File: strix/data_checker.py [hunk 1]
```diff
@@ -192,188 +91,40 @@ def get_train_valid_datasets():
...
-    channel, shape = data_shape[0], data_shape[1:]
-    logger.info(f"Data Channel: {channel}, Shape: {shape}")
+    if isinstance(train_dataset, torch.utils.data.DataLoader):
+        train_dataloader = train_dataset
+        valid_dataloader = valid_dataset
+    else:
+        train_num = min(cargs.n_batch, len(train_dataset))
+        valid_num = min(cargs.n_batch, len(valid_dataset))
+        train_dataloader = DataLoader(train_dataset, num_workers=1, batch_size=train_num, shuffle=True)
+        valid_dataloader = DataLoader(valid_dataset, num_workers=1, batch_size=valid_num, shuffle=False)
```
Comment (ldxFAIRYTAIL) [inline strix/data_checker.py:104]: Maybe we don't have to split train set, valid set and test set while checking data even, just check all the input data.

File: strix/data_checker.py [hunk 2]
```diff
@@ -192,188 +91,40 @@ def get_train_valid_datasets():
...
-                        data[f"{img_key}_meta_dict"],
-                        cargs.out_dir,
-                        phase,
-                        cargs.data_list,
-                        i,
-                        logger_name
-                    )
-
-                save_fnames(data, img_key + "_meta_dict", output_fpath)
+    overlap_m = "mask" if cargs.mask_overlap else "contour" if cargs.contour_overlap else None
```
Comment (ldxFAIRYTAIL) [inline strix/data_checker.py:109]: How about set `mask-overlap` as the default `overlap_m`?

File: strix/data_checker.py (hunk 1 above)
Comment (ChenglongWang) [reply strix/data_checker.py:104]: Users may have different processing for training, validation, test dataset.
